File: DDD5_UI/src/views/cms/publisher/publisher/publisherService.js

import axios from 'axios';

/**
 * 发布者信息
 */
export class PublisherService {
    // 分页查询列表
    static findPublishers (queryParams) {
        return axios({
            url: '/cms/publisher/Publisher/findPublishers',
            method: 'POST',
            data: queryParams
        });
    }

    //根据ID查询指定的发布者信息
    static findPublisher (publisherId) {
        const query = {
            publisherId
        }
        return axios({
            url: '/cms/publisher/Publisher/findPublisher',
            method: 'POST',
            params: query,
        });
    }

    //根据ID查询指定的发布者信息(包含外键名称)
    static findPublisherForView (publisherId) {
        const query = {
            publisherId
        }
        return axios({
            url: '/cms/publisher/Publisher/findPublisherForView',
            method: 'POST',
            params: query,
        });
    }

    //根据ID查询指定的发布者信息(包含外键名称)
    static findPublisherForEdit (publisherId) {
        const query = {
            publisherId
        }
        return axios({
            url: '/cms/publisher/Publisher/findPublisherForEdit',
            method: 'POST',
            params: query,
        });
    }

    //根据ID查询指定的发布者信息(只提取ID 和 Name)
    static findPublishersWithIdNameById (publisherId) {
        const query = {
            publisherId
        }
        return axios({
            url: '/cms/publisher/Publisher/findPublishersWithIdNameById',
            method: 'POST',
            params: query,
        });
    }

    //根据名称查询发布者信息集合(只提取ID 和 Name)
    static findPublishersWithIdNameByName (publisherName) {
        const query = {
            publisherName
        }
        return axios({
            url: '/cms/publisher/Publisher/findPublishersWithIdNameByName',
            method: 'POST',
            params: query,
        });
    }

    //新增保存发布者信息
    static savePublisher (publisher) {
        const query = {}
        return axios({
            url: '/cms/publisher/Publisher/savePublisher',
            method: 'POST',
            params: query,
            data: publisher
        });
    }

    //修改保存发布者信息
    static updatePublisher (publisher) {
        const query = {}
        return axios({
            url: '/cms/publisher/Publisher/updatePublisher',
            method: 'POST',
            params: query,
            data: publisher
        });
    }

    //根据ID查询指定的发布者信息(只提取ID 和 Name)
    static deletePublisher (publisherId) {
        const query = {
            publisherId
        }
        return axios({
            url: '/cms/publisher/Publisher/deletePublisher',
            method: 'POST',
            params: query,
        });
    }
}

File: DDD5/ems/src/main/java/com/mt/cms/service/publisher/PublisherService.java
package com.mt.cms.service.publisher;

import com.mt.common.core.web.base.PageDTO;
import com.mt.common.core.web.base.PageResultDTO;
import com.mt.cms.entity.publisher.Publisher;

import java.util.List;

public interface PublisherService {
    /**
     * 根据分页参数查询发布者信息集合
     *
     * @param pageDTO 分页条件
     */
    public PageResultDTO findPublishers(PageDTO pageDTO);

    /**
     * 查询全部发布者信息集合
     *
     */
    public List<Publisher> findAllPublishers();

    /**
     * 根据名称查询发布者信息集合(只提取ID 和 Name)
     *
     * @param publisherName 名称
     */
    public List<Publisher> findPublishersWithIdNameByName(String publisherName);

    /**
     * 查询所有发布者信息集合(只提取ID 和 Name)
     *
     */
    public List<Publisher> findAllPublishersWithIdName();

    /**
     * 根据ID查询指定的发布者信息(只提取ID 和 Name)
     *
     * @param publisherId Id
     */
    public Publisher findPublishersWithIdNameById(Long publisherId);

    /**
     * 根据ID查询指定的发布者信息
     *
     * @param publisherId Id
     */
    public Publisher findPublisher(Long publisherId);

    /**
     * 根据ID查询指定的发布者信息(包含外键)
     *
     * @param publisherId Id
     */
    public Publisher findPublisherWithForeignName(Long publisherId);

    /**
     * 新增发布者信息
     *
     * @param publisher 实体对象
     */
    public Publisher savePublisher(Publisher publisher);

    /**
     * 更新发布者信息
     *
     * @param publisher 实体对象
     */
    public Publisher updatePublisher(Publisher publisher);

    /**
     * 根据ID删除发布者信息
     *
     * @param publisherId ID
     */
    public void deletePublisher(Long publisherId);
}
